mantidproject/mantid
# Catch exception when invalid bgsub params passed in EngDiff UI fitting tab

**Description of work.**

Catch exception when invalid bgsub params passed in EngDiff UI fitting tab - in this instance set background to zero with zero errors.

I've had a look to see if I can add validation to NIter box to ensure only positive integers can be entered but it doesn't seem there's an easy way to do this...does anyone else know a way?

**To test:**

(1) Load data following the instructions in the manual testing docs (see Test 1 & 4)
https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html

(2) Follow the instruction in Test 6 and set NIter to -1, try setting XWindow to 1

Fixes #32353 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
